### PR TITLE
Aws api module add sts

### PR DESCRIPTION
**Pull request by mrcoggsworth**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: N/A

## Description
My goal is to help open up the AWSApiModule to be able to use session tokens as part of the class build.
The reason for this is due to using third-party authenticators such as Hashicorp vault sends back the end-user the
group of credentials in AWS. Adding the session token option directly to the helper class would ease additional repeated code inside each new integration submitted. This also allows for more control over client sessions for Cortex Xsoar servers not hosted on AWS infrastructure.

## Screenshots
N/A

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by mrcoggsworth** [comment]
All tests past, excited to make a first contribution and hopefully many more. Please let me know if anything needs to change/add, etc.

File: Packs/AWS-ACM/ReleaseNotes/1_1_8.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - ACM
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-ACM/ReleaseNotes/1_1_8.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-CloudWatchLogs/ReleaseNotes/1_2_4.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - CloudWatchLogs
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-CloudWatchLogs/ReleaseNotes/1_2_4.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-EC2/ReleaseNotes/1_2_9.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - EC2
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-EC2/ReleaseNotes/1_2_9.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-GuardDuty/ReleaseNotes/1_2_4.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - GuardDuty
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-GuardDuty/ReleaseNotes/1_2_4.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-IAM/ReleaseNotes/1_1_10.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - Identity and Access Management
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-IAM/ReleaseNotes/1_1_10.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-Lambda/ReleaseNotes/1_2_9.md [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - Lambda
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-Lambda/ReleaseNotes/1_2_9.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-S3/ReleaseNotes/1_2_1.md [hunk 7]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - S3
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-S3/ReleaseNotes/1_2_1.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-SQS/ReleaseNotes/1_2_5.md [hunk 8]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - SQS
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-SQS/ReleaseNotes/1_2_5.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

File: Packs/AWS-SecurityHub/ReleaseNotes/1_1_9.md [hunk 9]
```diff
@@ -0,0 +1,5 @@
+
+#### Integrations
+##### AWS - Security Hub
+- AWSClient Class in ApiModules/AWSApiModule now supports passing an AWS STS session token as a parameter
+- Allows for third-party authorization software such as Hashicorp Vault to provide login credentials to use for building a client session
```
Comment (DeanArbel) [inline Packs/AWS-SecurityHub/ReleaseNotes/1_1_9.md:5]:
```suggestion
- AWSClient class now supports passing an AWS STS session token as a parameter. This allows for third-party authorization software to provide login credentials to use for building a client session.
```

**Review by DeanArbel — COMMENTED** [review]
Looks good, I changed a bit the release notes. Please make sure to accept the suggestions.

**Review by ilaner — APPROVED** [review]
Great job

**Pull request merged**